### PR TITLE
Add extended diagnostic output from Thompson MP + regression test, includes "Add optional scaling to RRTMGP flux adjustment" (#605), bug fix in several suite definition files

### PR DESCRIPTION
**Pull request by climbfuji**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

This PR contains the following changes:
- Update submodule pointer for fv3atm and ccpp-physics for adding extended diagnostic output from Thompson MP
- Add a new regression test that exercises this capability
- Adjust the formatting in `rt.conf` and `rt_gnu.conf` so that the tables align
 
A new `diag_table` is required for the new test, this file can be added to the existing input data directory. 

**Note.** Because of the code changes in Thompson MP, the regression tests using Thompson with Intel in PROD mode change results (an optimization difference). With Intel in DEBUG mode, as well as with GNU in PROD and DEBUG mode, the results for the Thompson MP runs are identical.

This PR includes the changes in #605, "Add optional scaling to RRTMGP flux adjustment" - these will require an update of the RRTMGP baselines.

The associated fv3atm PR also includes bugfixes to several suite definition files (add missing calls to `GFS_radiation_surface`) that will change the answer of the regression tests using those suites.

### Issue(s) addressed

Fixes https://github.com/NCAR/ccpp-physics/issues/680.

### Initial testing

1. Create new baselines for the following tests, then verify against them on Hera using GNU and Intel: all tests pass.
```
control_thompson_debug
control_thompson_no_aero_debug
control_thompson_extdiag_debug (new test)
```
[rt_min_hera_gnu_create.log](https://github.com/ufs-community/ufs-weather-model/files/6697465/rt_min_hera_gnu_create.log)
[rt_min_hera_gnu_verify.log](https://github.com/ufs-community/ufs-weather-model/files/6697467/rt_min_hera_gnu_verify.log)
[rt_min_hera_intel_create.log](https://github.com/ufs-community/ufs-weather-model/files/6697469/rt_min_hera_intel_create.log)
[rt_min_hera_intel_verify.log](https://github.com/ufs-community/ufs-weather-model/files/6697471/rt_min_hera_intel_verify.log)

2. Verify against existing baselines on Hera using GNU and Intel: Because of the code changes in Thompson MP, the regression tests using Thompson with Intel in PROD mode change results (an optimization difference). With Intel in DEBUG mode, as well as with GNU in PROD and DEBUG mode, the results for the Thompson MP runs are identical. All other regression tests are b4b identical.

### Second regression testing (after merging in RRTMGP flux adjustment, fixing suite definition file bugs)

Verify against existing baselines on Hera using GNU and Intel: Because of the code changes in Thompson MP, RRTMGP and the suite definition file bugfixes, the following regression tests fail (because of b4b differences or missing baselines, they all run to completion):

With Intel:
```
control_thompson_extdiag_debug
control_rrtmgp_debug          
fv3_rrfs_v1alpha_debug        
control_rrtmgp                
control_thompson              
control_thompson_no_aero      
fv3_rrfs_v1alpha              
fv3_gsd                       
fv3_HAFS_v0_hwrf_thompson     
fv3_esg_HAFS_v0_hwrf_thompson 
regional_quilt_RRTMGP         
cpld_bmark_wave_v16_noahmp    
```
[rt_hera_intel_verify_against_existing.log](https://github.com/ufs-community/ufs-weather-model/files/6757692/rt_hera_intel_verify_against_existing.log)
[rt_hera_intel_verify_against_existing_fail_test.log](https://github.com/ufs-community/ufs-weather-model/files/6757693/rt_hera_intel_verify_against_existing_fail_test.log)

With GNU:
```
control_thompson_debug        
control_thompson_extdiag_debug
control_rrtmgp_debug          
control_rrtmgp                
```
[rt_hera_gnu_verify_against_existing.log](https://github.com/ufs-community/ufs-weather-model/files/6757701/rt_hera_gnu_verify_against_existing.log)
[rt_hera_gnu_verify_against_existing_fail_test.log](https://github.com/ufs-community/ufs-weather-model/files/6757702/rt_hera_gnu_verify_against_existing_fail_test.log)

### Final regression testing (at commit time)

**NOT YET APPLICABLE**

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] CI

## Dependencies

https://github.com/NCAR/ccpp-physics/pull/679
https://github.com/NOAA-EMC/fv3atm/pull/331
https://github.com/ufs-community/ufs-weather-model/pull/658

### CODE REVIEW

File: tests/fv3_conf/control_run.IN [hunk 1]
```diff
@@ -80,7 +80,8 @@ cp    @[INPUTDATA_ROOT]/${inputdir}/field_table_gfsv16 ./field_table
 cp    @[INPUTDATA_ROOT]/${inputdir}/*configure .
 
 if [ $NEW_DIAGTABLE != '' ]; then
-  cp    @[INPUTDATA_ROOT]/${inputdir}/$NEW_DIAGTABLE ./diag_table
+  #cp    @[INPUTDATA_ROOT]/${inputdir}/$NEW_DIAGTABLE ./diag_table
```
Comment (climbfuji) [inline tests/fv3_conf/control_run.IN:89]: This change is temporary until the new file `diag_table_gfsv16_thompson_extdiag` has been added to the input data directory.

**Comment by ericaligo-NOAA** [comment]
Looks good.

**Review by ericaligo-NOAA — APPROVED** [review]
With the 3D Thompson diagnostics reset by avg_max_length, all changes look good.

**Comment by climbfuji** [comment]
This PR was merged as part of https://github.com/ufs-community/ufs-weather-model/pull/619, including reverting the temporary change to the run scripts and adding the new diag_table for the extended Thompson microphysics run to the input data.

**Pull request closed without merging**